test/legacy_test/test_index_fill.py

```diff
@@ -0,0 +1,174 @@
+#   Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from itertools import combinations
+
+import numpy as np
+
+import paddle
+from paddle.base import Program
+
+paddle.enable_static()
+
+
+def compute_index_put_ref(x, axis, index, value):
```

Comment:
`compute_index_fill_ref` is more suitable?

Reply:
Is the logic of `compute_index_put_ref` too complex? and it is better to use Numpy's fancy indexing, eg, `x=np.transpose(x, perm); x[index] = value; x=np.transpose(x, perm)` be simpler and easier to understand?

Reply:
Done